base/compiler/ssair/legacy.jl

```diff
@@ -60,15 +56,21 @@ Mainly used for testing or interactive use.
 inflate_ir(ci::CodeInfo, linfo::MethodInstance) = inflate_ir!(copy(ci), linfo)
 inflate_ir(ci::CodeInfo, sptypes::Vector{VarState}, argtypes::Vector{Any}) = inflate_ir!(copy(ci), sptypes, argtypes)
 function inflate_ir(ci::CodeInfo)
-    slottypes = ci.slottypes
-    argtypes = Any[ slottypes === nothing ? Any : slottypes[i] for i = 1:length(ci.slotflags) ]
+    parent = ci.parent
+    isa(parent, MethodInstance) && return inflate_ir(ci, parent)
+    # XXX the length of `ci.slotflags` may be different from the actual number of call
+    # arguments, but we really don't know that information in this case
```

Comment:
Is this a new problem?

Reply:
Kind of. Previously we just created `ir.argtypes` based on the length of `ci.slotflags` no matter if `ci.slotflags` contains flags for local variables. That wasn't a problem, because the optimization passes don't care if `ir.argtypes` contains extra type information (and this is why we have used `ci.slottypes` as is for `ir.argtypes` in the main optimization pipeline).

After this commit, this code becomes a bit problematic since we will make additional efforts to make `ir.argtypes` contain types of call arguments only. Having said that, it's not that problematic usually (because the optimization pipeline doesn't care it currently). It would be problematic if we generate `OpaqueClosure` with `ir` created by `inflate_ir(ci)` with un-inferred `ci`, which triggers this code path, but it's such a rare case anyway.